test/test_crc32c.py

```diff
@@ -106,6 +107,49 @@ def assert_msvc_vals(b: bytes, crc: int, expected_crc: int) -> None:
         assert_msvc_vals(ulonglong_as_bytes(0x88889999EEEE3333), 0x5555AAAA, 0x16F57621)
 
 
+@pytest.mark.skipIf(crc32c is None, reason='no crc32c support in this platform')
+class TestCrc32cHash:
+    def test_misc(self) -> None:
+        crc32c_hash = crc32c.Crc32cHash(b"")
+
+        assert crc32c_hash.digest_size == 4
+        assert crc32c_hash.name == "crc32c"
+        assert len(crc32c_hash.digest()) == crc32c_hash.digest_size
+
+    def test_copy(self) -> None:
+        crc32c_hash = crc32c.Crc32cHash(b"")
+        crc32c_hash_copy = crc32c_hash.copy()
+
+        assert crc32c_hash.digest() == crc32c_hash_copy.digest()
+        assert id(crc32c_hash) != id(crc32c_hash_copy)
+
+        crc32c_hash.update(b"1")
+        crc32c_hash_copy.update(b"2")
+
+        assert crc32c_hash.digest() != crc32c_hash_copy.digest()
+
+    @pytest.mark.parametrize("data,digest,hexdigest,base64", [
+        (b"", b"\x00\x00\x00\x00", "00000000", "AAAAAA=="),
+        (b"23456789", b"\xbf\xe9\x2a\x83", "bfe92a83", "v+kqgw=="),
+        (b"123456789", b"\xe3\x06\x92\x83", "e3069283", "4waSgw=="),
```

Comment:
>take the results from the hash object's digest and hexdigest and turn them into the integer value that we know about already.

This is exactly what I *did not* want to do here. It's already tested (in multiple ways) that we are able to calculate the "right" CRC32C hash values of these inputs, as you said.

What this test explicitly tests is that the *representation* of these hashes is what we'd expect. This means for example that the padding with zeroes is critical, which you'd loose if you would just convert to an integer. Also, testing the exact base64 encoding here is critical.

What one could do is though is to give these `numbers1`, ... tuples 3 additional elements, i.e. `digest`, `hexdigest` and `base64` encoding. These would just be ignored in the existing tests but used in these new tests.

Reply:
> It's already tested [...] that we are able to calculate the "right" CRC32C hash values of these inputs

Indeed, and here we wouldn't be testing the validity of the checksums really, but rather the representation of the digest/hexdigest against the numerical value.

> for example that the padding with zeroes is critical, which you'd loose if you would just convert to an integer

You can assert on the length of the bytes (you already do in one of the tests, but maybe you can do the same in this parametrised test), and then again that the bytes -> int conversion is the expected one. That would cover any padding issues I believe.

> the exact base64 encoding here is critical

As mentioned before, base64 is something I'd rather we don't add for the time being. In any case, I find it more critical that the bytes are the correct ones, since the base64 encoding's main input are those bytes.

Reply:
Done. Although I have to say I still prefer explicitness, even over slight code/value repetition, because there simply is no room for errors then, which can be important for a library like this.